lib/scrub/chunker.rb

```diff
@@ -9,31 +10,47 @@ module Scrub
   # #{chunk_count} equal-sized (roughly) files from them.
   class Chunker
     attr_reader :glob, :chunks
-    def initialize(glob, chunk_count = 16)
+    def initialize(glob: "/dev/null", chunk_count: 16, add_uuid: false, out_ext: nil)
       @glob = glob # to one or more files
       @chunks = [] # store paths to output files here
       @chunk_count = chunk_count
+      @add_uuid = add_uuid
       @work_dir = Settings.scrub_chunk_work_dir
+
       if @work_dir.nil?
-        raise "Settings.scrub_chunk_work_dir must be set, is nil"
+        raise ArgumentError, "Missing Settings.scrub_chunk_work_dir"
       end
+
       @uuid = SecureRandom.uuid
-      FileUtils.mkdir_p("#{@work_dir}/#{@uuid}")
+      @out_ext = out_ext # Apply this extension to the resulting file(s).
+      @tmp_chunk_dir = File.join(@work_dir, @uuid)
+      FileUtils.mkdir_p(@tmp_chunk_dir)
+    end
+
+    def validate
+      if @work_dir.nil?
+        raise Scrub::ChunkerError, "Settings.scrub_chunk_work_dir must be set, is nil"
+      end
+      if @out_ext.nil?
+        raise Scrub::ChunkerError, "No output extension specified (@out_ext)"
+      end
     end
 
     def run
       # First we need to get all the data from @glob into one file.
       # `split --number=l/x` only works on files, not on STDIN.
       # TODO: we might want another dir for sort -T .
-      tmp_file = "#{@work_dir}/#{@uuid}/tmp_sorted.txt"
+      tmp_file = File.join(@tmp_chunk_dir, "tmp_sorted.txt")
+      add_uuid_call = @add_uuid ? "| bundle exec ruby bin/add_uuid.rb" : ""
       sort_call = "egrep -vh '^OCN' #{@glob} | " \
-                  "sort -s -n -k1,1 -T ./ | " \
-                  "bundle exec ruby bin/add_uuid.rb > #{tmp_file}"
+                  "sort -s -n -k1,1 -T ./ " \
+                  "#{add_uuid_call} " \
```

Comment:
I waffled a little bit about how I felt about the external calls here. Ultimately, I think it's OK; otherwise, there would have to be more temporary files or complicated pipe logic, and add_uuid.rb is pretty simple.

Reply:
Yeah I did not feel great writing this.